nmdc_runtime/api/endpoints/util.py

```diff
@@ -57,9 +58,25 @@
 HOSTNAME_EXTERNAL = BASE_URL_EXTERNAL.split("://", 1)[-1]
 
 
+def check_filter(filter_: str):
+    if not filter_.startswith("{") or not filter_.endswith("}"):
```

Comment:
I don't know whether you want to allow the `filter` value to begin or end with whitespace. If so, I'd suggest:

```diff
- if not filter_.startswith("{") or not filter_.endswith("}"):
+ if not filter_.lstrip().startswith("{") or not filter_.rstrip().endswith("}"):
```

Docs:
- `lstrip()` - https://docs.python.org/3/library/stdtypes.html#str.lstrip

Reply:
cool. I added `filter_ = filter_.strip()` at the top of `check_filter` because I think it's fine to tolerate extra whitespace on the ends for user input.